plugins/lsp/conform-nvim.nix

```diff
@@ -0,0 +1,89 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.conform-nvim;
+  helpers = import ../helpers.nix {inherit lib;};
+in {
+  options.plugins.conform-nvim =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "conform-nvim";
+
+      package = helpers.mkPackageOption "conform-nvim" pkgs.vimPlugins.conform-nvim;
+
+      # Map of filetype to formatters
+      formattersByFt =
+        helpers.defaultNullOpts.mkNullable types.attrs
+        "see documentation"
+        ''
+          Map of filetype to formatters
+          lua = { "stylua" },
+          Conform will run multiple formatters sequentially
+          python = { "isort", "black" },
+          Use a sub-list to run only the first available formatter
+          javascript = { { "prettierd", "prettier" } },
+          Use the "*" filetype to run formatters on all filetypes.
+          ["*"] = { "codespell" },
+          Use the "_" filetype to run formatters on filetypes that don't
+          have other formatters configured.
+          ["_"] = { "trim_whitespace" },
+        '';
+
+      # If this is set, Conform will run the formatter on save.
+      # It will pass the table to conform.format().
+      # This can also be a function that returns the table.
+      # See :help conform.format for details.
```

Comment:
This would be better either as the documentation of the options, or by converting formatOnSave to a sub-module and setting it to the sub-module docmentation

Reply:
I did try to define a type for it but I failed. I think using `attrsOf` might not be correct although we can add a description this way:
```
      formatOnSave =
        helpers.defaultNullOpts.mkNullable types.attrsOf types.anything
```

Reply:
You need to do `(types.attrsOf types.anything)`

Reply:
But there is a `types.submodule` where you can do what you want`

Example:
```nix
formatOnSave = helpers.defaultNullOpts.mkNullable (types.submodule {
    options = { #options... };
};
```